go/ql/lib/semmle/go/dataflow/internal/DataFlowUtil.qll

```diff
@@ -376,13 +376,8 @@ module BarrierGuard<guardChecksSig/3 guardChecks> {
         // Case: a function like "return someBarrierGuard(arg)"
         // or "return !someBarrierGuard(arg) && otherCond(...)"
         exists(boolean outcome |
-          ret = getUniqueOutputNode(fd, outp) and
```

Comment:
To check: is the standard pipeline (cf the base pipeline, used for iteration 1) also using a sensible join order?